orchestrator.py

```
@@ -179,6 +175,107 @@ class PreprocessedDocument:
+        """Normalize arbitrary ingestion payloads into orchestrator documents."""
+
+        if isinstance(document, cls):
+            return document
+
+        if hasattr(document, "raw_document") and hasattr(document, "full_text"):
+            return cls._from_ingestion(document, document_id=document_id)
+
+        raise TypeError(
+            "Unsupported preprocessed document payload: "
+            f"expected orchestrator or document_ingestion schema, got {type(document)!r}"
+        )
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)